import winstonLogger from './winston';

process.on('uncaughtException', (err) => {
  try {
    console.error('ðŸ”´ [ Error ] Uncaught Exception caught', err);
    winstonLogger.error(`ðŸ”´ [ Error ] Uncaught Exception caught ${err}`);
  } catch (error) {
    console.error('ðŸ”´ [ Error ] Uncaught Exception caught', error);
  }
});

process.on('unhandledRejection', (reason, promise) => {
  try {
    console.error(
      'ðŸ”´ [ Error ] Unhandled Rejection caught',
      promise,
      'reason:',
      reason,
    );
    winstonLogger.error(
      `ðŸ”´ [ Error ] Unhandled Rejection caught ${promise} reason: ${reason}`,
    );
  } catch (error) {
    console.error('ðŸ”´ [ Error ] Unhandled Rejection caught', error);
  }
});
